backend/pkg/plugins/plugins.go

```diff
@@ -52,6 +52,10 @@ func periodicallyWatchSubfolders(watcher *fsnotify.Watcher, path string, interva
 	for ; true; <-ticker.C {
 		// Walk the path and add any new directories to the watcher.
 		_ = filepath.WalkDir(path, func(path string, d fs.DirEntry, err error) error {
+			if strings.Contains(path, ".plugins") {
+				return filepath.SkipDir
+			}
```

Comment:
This function (periodicallyWatchSubfolders) should be fine. The problem is that we are including from a higher level the static plugins dir. The static plugins dir should be included for building Headlamp with the shipped plugins, but it's called static because we don't expect those to be changed, and thus we should not listen to changes in the the static folder (whatever that folder is; nowadays it's by default .plugins, but could be something else).

Reply:
when running `make run-backend` locally it shows the .plugins folder as the plugins dir rather than the static plugins dir, so would it make sense to only watch for changes to the plugin dir when there are plugins defined in static plugin dir?

Reply:
That's unrelated IIUC. If we are picking up .plugins/ as taking precedence (over the default folder), then that should be changed, but that's a different PR. This one should be only about not watching the static plugins dir for changes.